core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockResolver.scala

```diff
@@ -41,6 +41,14 @@ trait ShuffleBlockResolver {
    */
   def getBlockData(blockId: BlockId, dirs: Option[Array[String]] = None): ManagedBuffer
 
+  /**
+   * Retrive a list of BlockIds for a given shuffle map. Used to delete shuffle files
+   * from the external shuffle service after the associated executor has been removed.
+   */
+  def getBlocksForShuffle(shuffleId: Int, mapId: Long): Seq[BlockId] = {
+    Seq.empty
+  }
```

Comment:
I wasn't sure whether I should make this backward compatible or not, so I added a default implementation. I can remove it if it doesn't need to be backward compatible (which it's marked private so probably doesn't need to be?)

Reply:
I agree with your assessment @Kimahriman and would prefer to avoid this methough, though @Ngone51 had expressed concern in the past that some of the private classes might be getting used for some custom implementations.
Thoughts @Ngone51  ?